virttest/qemu_vm.py

```diff
@@ -448,9 +448,19 @@ def add_serial(devices, name, filename):
             if '86' in params.get('vm_arch_name', arch.ARCH):
                 cmd += " -device isa-serial"
             elif 'ppc' in params.get('vm_arch_name', arch.ARCH):
+                if ('tcp'in params.get('pro_backend')):
```

Comment:
The socket protocol should not be related to architecture, thus the right way to implement this could be:
```python
def add_serial(devices, name, filename):
    ...
    serial_id = "serial_id_%s" % name
    cmd = " -chardev socket"
    cmd += _add_option("id", serial_id)
    if params.get('pro_backend', 'unix')) == 'tcp':
        cmd += _add_option("host", '0.0.0.0')
        cmd += _add_option("port", str(utils_misc.find_free_port(12000, 13000)))
    else:
        cmd += _add_option("path", filename)
    cmd += _add_option("server", "NO_EQUAL_STRING")
    cmd += _add_option("nowait", "NO_EQUAL_STRING")
    if '86' in params.get('vm_arch_name', arch.ARCH):
        ...
    elif 'ppc' in params.get('vm_arch_name', arch.ARCH):
        ...
```
Also, the param name `pro_backend` is not good, I suggest to use `serial_sockproto` instead.